### PR TITLE
`Table` - Add example with `rowspan/colspan` to showcase

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

Context: https://hashicorp.slack.com/archives/C7KTUHNUS/p1700162017441919

### :hammer_and_wrench: Detailed description

In this PR I have:
- added to the showcase an example of an `Hds::Table` with `rowspan/colspan`attributes 

### :camera_flash: Screenshots

screenshot_3270

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Review by alex-ju — APPROVED** [review]
Out of curiosity – do our designers have any thoughts on what cells spanning multiple rows should look like? (e.g. we could have vertical borders only for these cells to make it look less broken)

**Comment by didoo** [comment]
> Out of curiosity – do our designers have any thoughts on what cells spanning multiple rows should look like? (e.g. we could have vertical borders only for these cells to make it look less broken)

@hashicorp/hds-design thoughts?

**Comment by KristinLBradley** [comment]
The added example looks good to me but I would like to see input from design on the visual style first before merging.

**Comment by didoo** [comment]
> Out of curiosity – do our designers have any thoughts on what cells spanning multiple rows should look like? (e.g. we could have vertical borders only for these cells to make it look less broken)

> The added example looks good to me but I would like to see input from @hashicorp/hds-design on the visual style first before merging.

@alex-ju @KristinLBradley here's @heatherlarsen comment: "In that case [it's only for the showcase], I think it’s fine to merge the PR. But I’ll create a ticket so we can come up with a better visual solution for this."

I think it's safe to merge at this point.

**Comment by jorytindall** [comment]
Noted that the ticket has been created for addressing this in the future, but I think ideally there would be some visual differentiation between the rows and the spanning column. I'll defer to others, but if there's time it would be nice to include this in the design work for the the current table work/enhancements.

**Comment by didoo** [comment]
> Noted that the ticket has been created for addressing this in the future, but I think ideally there would be some visual differentiation between the rows and the spanning column.

@hashicorp/hds-design I don't see how this is possible without introducing vertical lines for the cells. Should we introduce an option for consumers to have these vertical lines too, and provide guidance to use them only for specific use cases?

> I'll defer to others, but if there's time it would be nice to include this in the design work for the the current table work/enhancements.

Since we're working on the visual appearence of the table headers, and adding vertical lines impacts them visually, I would opt for a quick (quick!) exploration also of this option. As I remember, there were already explorations of this variant at the beginning, it should not be too complex (I _think_).

/cc @hashicorp/hds-engineering @Dhaulagiri

**Pull request merged**
